Modules/System/Rest Client/src/HttpContent.Codeunit.al

```diff
@@ -0,0 +1,279 @@
+    /// <remarks>The Content-Type header will be set to 'application/json'.</remarks>
+    procedure Create(Content: JsonObject) HttpContent: Codeunit "Http Content"
+    begin
+        SetContent(Content);
+        HttpContent.SetHttpContentImpl(HttpContentImpl);
+    end;
+
+    /// <summary>Initializes a new instance of the Http Content class with the specified JsonArray content.</summary>
+    /// <param name="Content">The content to send to the server.</param>
+    /// <returns>The Http Content object.</returns>
+    /// <remarks>The Content-Type header will be set to 'application/json'.</remarks>
+    procedure Create(Content: JsonArray) HttpContent: Codeunit "Http Content"
+    begin
+        SetContent(Content);
+        HttpContent.SetHttpContentImpl(HttpContentImpl);
+    end;
+
+    /// <summary>Initializes a new instance of the Http Content class with the specified JsonToken content.</summary>
+    /// <param name="Content">The content to send to the server.</param>
+    /// <returns>The Http Content object.</returns>
+    /// <remarks>The Content-Type header will be set to 'application/json'.</remarks>
+    procedure Create(Content: JsonToken) HttpContent: Codeunit "Http Content"
+    begin
+        SetContent(Content);
+        HttpContent.SetHttpContentImpl(HttpContentImpl);
+    end;
+
+    /// <summary>Initializes a new instance of the Http Content class with the specified XmlDocument content.</summary>
+    /// <param name="Content">The content to send to the server.</param>
+    /// <returns>The Http Content object.</returns>
+    /// <remarks>The Content-Type header will be set to 'text/xml'.</remarks>
+    procedure Create(Content: XmlDocument) HttpContent: Codeunit "Http Content"
+    begin
+        SetContent(Content);
+        HttpContent.SetHttpContentImpl(HttpContentImpl);
+    end;
+
+    /// <summary>Initializes a new instance of the Http Content class with the specified "Temp Blob" Codeunit content.</summary>
+    /// <param name="Content">The content to send to the server.</param>
+    /// <returns>The Http Content object.</returns>
+    /// <remarks>The Content-Type header will be set to 'application/octet-stream'.</remarks>
+    procedure Create(Content: Codeunit "Temp Blob") HttpContent: Codeunit "Http Content"
+    begin
+        HttpContent := Create(Content, '');
+    end;
+
+    /// <summary>Initializes a new instance of the Http Content class with the specified "Temp Blob" Codeunit content and content type.</summary>
+    /// <param name="Content">The content to send to the server.</param>
+    /// <param name="ContentType">The content type of the content to send to the server.</param>
+    /// <returns>The Http Content object.</returns>
+    /// <remarks>If the ContentType parameter is not specified, it will be set to 'application/octet-stream'.</remarks>
+    procedure Create(Content: Codeunit "Temp Blob"; ContentType: Text) HttpContent: Codeunit "Http Content"
```

Comment:
I think this procedure is missing documentation on the Default Encoding UTF8 that is used to read the tempBlob.
Maybe an overload where we can specify the Encoding might be useful.

Reply:
Good suggestion. I've chosen to get this PR processed and into the main branch, as we've got a few other PRs waiting to uptake this one. @ajkauffmann, you might want to consider this comment for a follow-up PR (once we switch to the main branch, where you'll find your submitted module 🥳).